cmd/crc/cmd/start.go

```diff
@@ -268,6 +276,9 @@ func writeOpenShiftTemplatedMessage(writer io.Writer, s *startResult) error {
 	if s.ClusterConfig.ClusterType == preset.OKD {
 		tmpl = fmt.Sprintf("%s\n\n%s", startTemplateForOpenshift, startTemplateForOKD)
 	}
+	if s.FallbackPortWarning != "" {
+		tmpl = fmt.Sprintf("%s\n%s", tmpl, fallbackPortWarning)
+	}
```

Comment:
Do we get any benefits from having `FallbackPortWarning` in `startResult` rather than calling `portFallbackWarning()` here?

Reply:
If it is not part of result then you will not get it as part of `crc start -ojson`, in case some other app want to make use of this warning to show then it can be used.
```
$ crc start -o json
[...]
   "developerCredentials": {
      "username": "developer",
      "password": "developer"
    }
  },
  "fallbackPortWarning": "\nNote: Port 8080 is Assigned.\nYou have to add this port to the application route to access\nit, such as http://myapp.apps-crc.testing:8080\n"
}

```

Reply:
If  we want to expose this in json,  this would look more useful than an arbitrary string:
```
[...]
   "developerCredentials": {
      "username": "developer",
      "password": "developer"
    }
  },
  "alternativeIngressPorts": {
      "http": "8080",
  }
}
```

Reply:
What we want to expose is debatable, warning can be useful if the app just need to popup or even show in the terminal. Also this is the same warning what crc start cmd shows. If we just put alternativeIngressPorts then user of it need to put own interpretation/warning around it.

Reply:
In my opinion, a preformatted string is always never the right thing to have in an API, be it a json struct, a function return value, ... As soon as you need something different from the hardcoded string, you have to add awkward parsing code to get back the original data, and rebuild what you need from it.
One way forward is to not add anything to the json output for now, and to add 'something' when we have clear requirements for it.

Reply:
@cfergeau Ack, I remove the warn string from json output now.